Fixed plugin log attributes

levb opened the pull request:
#### Summary
Unless I am mistaken, this should fix field names for structured logs (...keyValuePair). I noticed it because the plugin logs appeared broken.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-38342

wiggin77 commented:
Thanks @levb   Looks like I broke this when mlog was refactored.  I missed the fact that our SugarLogger wrapper was calling the key/value version of Zap's sugar logger https://github.com/uber-go/zap/blob/914c4ff628cdc5329a10e22b46b8e0697da23b0e/sugar.go#L172  I thought it was just providing something compatible with the standard logger.

To fix this some extra `XXXw` methods need to be added to the sugar logger, and a method to sweeten the fields list ala Zap.  Then the plugin API needs a minor change to call these new k/v methods.  

I can make those changes unless you'd rather do it.

levb commented:
@wiggin77 honestly, I have no appetite for fixing it all-around since I don't understand all the use-cases in the server code-base. In `mattermost-plugin-apps` I implemented another zap wrapper, wraps p.API.LogDebug etc (#onion), I just added my own provider, but had to define a `With` that returns an my type, not sugarlogger since it's concrete.  https://github.com/mattermost/mattermost-plugin-apps/blob/master/utils/zap_logger.go

wiggin77 commented:
Ok, @levb I'll work on the fix today.  I'm not entirely sure what your Zap wrapper is for... hopefully it won't be needed.

wiggin77 commented:
This issue is fixed via https://github.com/mattermost/mattermost-server/pull/18368 and latest Logr release.
